eth/transactionManager.go

```diff
@@ -234,30 +241,37 @@ func (tm *TransactionManager) checkTxLoop() {
 	}
 }
 
+func (tm *TransactionManager) newAdjustedTx(tx *types.Transaction) *types.Transaction {
+	baseTx := newDynamicFeeTx(tx)
+	if tm.gpm.MaxGasPrice() != nil {
+		baseTx.GasFeeCap = tm.gpm.MaxGasPrice()
```

Comment:
I forget if we talked about this, but is there a reason for always setting this to the max gas price as opposed to only setting it to the max gas price if the max gas price exceeds the current value?

Reply:
You're right. Actually, there is one more scenario for the replacement if the user manually increases `maxGasPrice` while the transaction is pending. Then the replacement can also happen.